File: java/ecbenchmark/src/ecbenchmark/function/Rastrigin.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.function;

import ecbenchmark.Function;
import java.util.Random;

/**
 *
 * @author jcrada
 */
public class Rastrigin implements Function {

    public Rastrigin() {
    }

    public double f(final double[] x) {
        double result = 0.0;
        for (int i = 0; i < x.length; ++i) {
            result += x[i] * x[i] - 10 * Math.cos(2 * Math.PI * x[i]) + 10;
        }
        return result;
    }
}

File: cpp/ecbenchmark/src/function/Rastrigin.cpp
#include "ecbenchmark/function/Rastrigin.h"

#include <math.h>
namespace ecb {
   /* Source:
    * @incollection{springer-dif.evo,
        Author = {Price, Kenneth and Storn, Rainer M. and Lampinen, Jouni A.},
        title={{Appendix}},
        booktitle = {Differential Evolution: A Practical Approach to Global Optimization},
        series={Natural Computing Series},
        editor={Rozenberg, G. and Bäck, Th. and Eiben, A. E. and Kok, J. N. and Spaink, H. P.},
        publisher = {Springer-Verlag Berlin Heidelberg},
        pages = {513-533},
        year = {2005}
        }
    * 
    * Range: -5.12 - 5.12
    */
    scalar Rastrigin::f(const std::vector<scalar>& x) {
        scalar result = 0.0;
        for (size_t  i = 0; i < x.size(); ++i) {
            result += x[i] * x[i] - 10 * cos(2 * M_PI * x[i]) + 10;
        }
        return result;
    }
}
